### DISCUSSION TITLE
AUTOMATION: Uploading HTML reports to s3 and providing the behave output-directory

### DISCUSSION THREAD

**Discussion by Akhilteegala** [behave/behave#1209] [Q&A] [answered]
Hi All,

I have a requirement to upload the local html reports to s3. How can we do it is there a way to handle this. every time I run behave I am specifying the output folder so how does the behave knows the exact folder to upload and what is the ideal way to do this.

I also have a requirement to show the output of the each step under the given when then tags in the html report. so that we don't have to go inside the logs to see what result I has given, is there a way to show the output on html report.

**Comment by jenisys** [comment, marked as answer]
## PART 1: Automation Problem
What you describe is a **build-script problem** and not a **behave problem**:

* You use some kind of build script to run behave with the provided output-directory
* After the behave run, you copy these files to the s3 filesystem (by using [s3fs] or …)

POTENTIAL SOLUTIONS:

* Use a shell script (as wrapper around the behave run)
* Use a build-system like `make`, [just], [invoke] or …

[just]: https://github.com/casey/just
[invoke]: https://pyinvoke.org
[s3fs]: https://github.com/fsspec/s3fs

## PART 2: HTML Formatter Problem

Just use a HTML Formatter for `behave` that supports your requirements.

CANDIDATES:

* [behave-html-pretty-formatter]
* [behave-html-formatter]
* [allure-behave] formatter
* Write your own customized HTML formatter (or derive it from one of the existing ones)

[behave-html-pretty-formatter]: https://github.com/behave-contrib/behave-html-pretty-formatter
[behave-html-formatter]: https://github.com/behave-contrib/behave-html-formatter
[allure-behave]: https://github.com/allure-framework/allure2
